packages/smooth_app/lib/pages/product/edit_ingredients_page.dart

```diff
@@ -129,18 +130,25 @@ class _EditOcrPageState extends State<EditOcrPage> {
     }
   }
 
-  Future<bool> _updateText(final String text) async {
-    final Product minimalistProduct =
-        _helper.getMinimalistProduct(_product, text);
+  Future<bool> _updateText(
+      final String text, UpToDateProductProvider provider) async {
+    final LocalDatabase localDatabase = context.read<LocalDatabase>();
+    final DaoProduct daoProduct = DaoProduct(localDatabase);
+    Product changedProduct = Product(barcode: _product.barcode);
+    Product? cachedProduct = await daoProduct.get(_product.barcode!);
+    if (cachedProduct != null) {
+      cachedProduct = _helper.getMinimalistProduct(cachedProduct, text);
```

Comment:
Unfortunately the `getMinimalistProduct` method name is a bit confusing but that's not your fault, it pre-existed.
After this PR, with a clear mind, we'll be able to add some coherence and refactor.